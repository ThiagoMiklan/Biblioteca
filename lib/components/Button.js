"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  disabled: 'boolean',
  "delete": 'boolean',
  definition: 'string',
  custom: 'object',
  onClick: 'function'
};

var Button = function Button(props) {
  (0, _type_validations.validate)(props, props_obj, "Button");
  var definition = getDefinition(props);
  return /*#__PURE__*/React.createElement("button", (0, _extends2["default"])({}, props.custom, {
    disabled: props.disabled === true ? true : false,
    className: definition,
    onClick: props.onClick
  }), props.label);
};

function getDefinition(props) {
  var definition = "";

  if (props["delete"] === true) {
    definition = "delete";
  } else {
    definition = props.definition === undefined ? "button" : "button " + props.definition;
  }

  return definition;
}

var _default = Button;
exports["default"] = _default;