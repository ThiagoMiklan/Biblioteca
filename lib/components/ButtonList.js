"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: "string",
  custom: 'object'
};

var ButtonList = function ButtonList(props) {
  (0, _type_validations.validate)(props, props_obj, "ButtonList");
  return assembleButtonList(props);
};

function assembleButtonList(props) {
  (0, _type_validations.validate)(props, props_obj, "ButtonList");
  var code_icon = "";
  var definition = props.definition == undefined ? "buttons" : "buttons " + props.definition;

  if (definition.includes("addons")) {
    code_icon = /*#__PURE__*/React.createElement("div", (0, _extends2.default)({
      className: props.definition
    }, props.custom), React.Children.map(props.children, function (child, i) {
      return /*#__PURE__*/React.createElement("p", {
        className: "control"
      }, /*#__PURE__*/React.createElement("div", {
        className: definition
      }, child));
    }));
  } else {
    code_icon = /*#__PURE__*/React.createElement("div", {
      className: definition
    }, props.children);
  }

  return code_icon;
}

var _default = ButtonList;
exports.default = _default;