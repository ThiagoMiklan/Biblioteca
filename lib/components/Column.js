"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: "string"
};

var Column = function Column(props) {
  (0, _type_validations.validate)(props, props_obj, "Column");
  var definition = props.definition == undefined ? "column" : "column " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, props.children);
};

var _default = Column;
exports.default = _default;