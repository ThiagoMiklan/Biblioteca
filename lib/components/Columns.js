"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: 'string'
};

var Columns = function Columns(props) {
  (0, _type_validations.validate)(props, props_obj, "Columns");
  var definition = props.definition == undefined ? "columns" : "columns " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, React.Children.map(props.children, function (child, i) {
    return child;
  }));
};

var _default = Columns;
exports["default"] = _default;