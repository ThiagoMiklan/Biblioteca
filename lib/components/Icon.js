"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

require("../css/all");

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: "string",
  right: "boolean",
  onClick: "function",
  icon_name: "string"
};

var Icon = function Icon(props) {
  (0, _type_validations.validate)(props, props_obj, "Icon");
  var definition = props.definition == undefined ? "icon" : "icon " + props.definition;
  return /*#__PURE__*/React.createElement(React.Fragment, null, props.right == false || props.right == undefined ? /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("span", {
    onClick: props.onClick,
    className: definition
  }, /*#__PURE__*/React.createElement("i", {
    className: props.icon_name
  })), /*#__PURE__*/React.createElement("span", null, props.children)) : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement("span", null, props.children), /*#__PURE__*/React.createElement("span", {
    onClick: props.onClick,
    className: definition
  }, /*#__PURE__*/React.createElement("i", {
    className: props.icon_name
  }))));
};

var _default = Icon;
exports.default = _default;