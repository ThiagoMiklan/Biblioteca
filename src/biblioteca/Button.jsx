// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import {renderToString} from 'react-dom/server';

type Props = {
  disabled?: bool,
  delete?: bool,
  definition?: string,
  children?: React.Node,
  custom?: any,
  onClick?: () => void
}

/*

Possui children
Eventos: onClick no <button> </button>
Observações: 

    1 - Caso exista a necessidade de adicionar um ícone ao componente Button, é necessário seguir
             a seguinte estrutura:

            <Button definition="info" >
                <Icon icon_name="fas fa-check" [right={true} OR empty prop right]>
                    Nome do botão
                </Icon>
            </Button >

            Sendo necessário informar o componente Icon como filho do componente Button
            e o nome do botão deve ser informado no Icon e não no Button como em outras
            situações onde usa-se o componente isolado.
*/

const Button= (props: Props)=>{
    var definition = getDefinition(props);
    
     return <button 
                {...props.custom} 
                disabled = {props.disabled == true? true: false } 
                className={definition} 
                onClick={props.onClick}
                >
                {props.children}
         </button>
         ;
}

function getDefinition(props){
    var definition = "";

    if(props.delete ==  true){
        definition = "delete";
    }else{
        definition = (props.definition == undefined) ? "button" : "button "+ props.definition;
    }

    return definition;
}

Button.propTypes = {
    definition: PropTypes.string,
    onClick: PropTypes.func,
    delete: PropTypes.bool,
    disabled: PropTypes.bool
};


export default Button;