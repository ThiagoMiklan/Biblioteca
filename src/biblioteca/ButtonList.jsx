// @flow
import * as React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassNames from '../tools/getClassName';


// Componente criado com o objetivo de servir de containe para diversos botões
// Oferece possibiidade de escolher se o container é "field" ou "buttons", para escolher field informar field={true}
// Class Field : Funcona com diversos elementos e não somente botões
// Class Buttons: Funciona apenas com botões

type Props ={
    definition?: string,
    field?: bool,
    children?:React.Node
}

const ButtonList = (props:Props) => {
    return (assembleButtonList(props));
}

function assembleButtonList(props) {
    var code_icon = "";
    var definition = getDefinition(props.definition, props.field);

    // permite o funcionamento dos addons, pois fornece uma interação que gera os filhos dentro de uma tag <p className=control>
    if (definition.includes("addons")) {
        code_icon =
            <div className={definition}>
                {
                    React.Children.map(props.children, (child, i) => {
                        return <p className="control">
                            <div className={definition}>
                                {child}
                            </div>
                        </p>
                    })
                }
             </div>

    } else {
        code_icon =
            <div className={definition}>
                {props.children}
            </div>;
    }

    return code_icon;
}

function getDefinition(definition:any, field) {

    if (field == true) {
        definition = classnames(getClassNames("field " + definition, "ButtonList"));;
    } else {
        definition = classnames(getClassNames("buttons " + definition, "ButtonList"));;
    }
    return definition;
}

ButtonList.propTypes = {
    definition: PropTypes.string,
    field: PropTypes.bool
};

export default ButtonList;