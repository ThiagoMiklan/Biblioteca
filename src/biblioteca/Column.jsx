import React from 'react';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import PropTypes from 'prop-types';
/**
 * Componente inspirado na classe column do bulma
 * Utilizado como filho do componente Columns
 * Possui filhos, pois funciona como um container para outros
 * elementos
 * Sem eventos
 */

const Column = (props) => {
    var definition = classnames(getClassName("column "+ props.definition, "Column"));
    return (
        <div className={definition}>{props.children}</div>
    );
}

Column.propTypes = {
    // caracter√≠sticas do bulma para a coluna
    definition : PropTypes.string
}

export default Column;