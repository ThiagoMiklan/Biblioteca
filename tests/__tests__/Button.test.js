import React from 'react';
import Button from '../../src/components/Button';
import ButtonList from '../../src/components/ButtonList';
import Icon from '../../src/components/Icon';
import renderer from 'react-test-renderer';
import Columns from '../../src/components/Columns';
import Column from '../../src/components/Column'


function onClick() {
  alert(testando);
}

test('value correctly', () => {
  const componentWithValue = renderer
    .create(<Button label="Value"/>)
    .toJSON();
  // componente com valor
  expect(componentWithValue).toMatchSnapshot();

  // componente sem valor
  const componentWithoutValue = renderer.create(<Button />)
    .toJSON();
  expect(componentWithoutValue).toMatchSnapshot();

});

// testa se as cores estão corretas, cores simples sem características adicionais
test('color correctly', () => {

  const simple_color = renderer.create(
  <ButtonList>
      <Button  definition="is-primary" label="Primary"/>
      <Button  definition="is-link" label="Link"/>
      <Button  definition="is-info" label="Info"/>
      <Button  definition="is-success" label="Success"/>
      <Button  definition="is-warning" label="Warning"/>
      <Button  definition="is-danger" label="Danger"/>
  </ButtonList>
    ).toJSON();

  expect(simple_color).toMatchSnapshot();
});

// testa se as cores estão corretas, cores secundárias dos botões
test('color secundary correctly', () => {

  const secundary_color = renderer.create(
    <ButtonList>
      <Button  definition="is-white" label="White"/>
      <Button  definition="is-light" label="Light"/>
      <Button  definition="is-dark" label="Dark"/>
      <Button  definition="is-black" label="Black"/>
      <Button  definition="is-text" label="Text"/>
  </ButtonList>).toJSON();

  expect(secundary_color).toMatchSnapshot();
});

// testa se as cores estão corretas, cores com característica claras
test('colors light correctly', () => {

  const light_color = renderer.create(
  <ButtonList>
    <Button  definition="is-primary is-light" label="primary light"/>
    <Button  definition="is-link is-light" label="link light"/>
    <Button  definition="is-info is-light" label="info light"/>
    <Button  definition="is-success is-light" label="success light"/>
    <Button  definition="is-warning is-light" label="warning light"/>
    <Button  definition="is-danger is-light" label="danger light"/>
  </ButtonList>).toJSON();

  expect(light_color).toMatchSnapshot();
});

// testa se os tamanhos dos botões estão corretos
test('buttons size correctly', () => {

  const light_color = renderer.create(
  <ButtonList>
    <Button definition="button is-small" label="Small"/>
    <Button label="Normal"/>
    <Button definition="button is-medium" label="Medium"/>
    <Button definition="button is-large" label="Large"/>
  </ButtonList>
    ).toJSON();

  expect(light_color).toMatchSnapshot();
});

// testa se os tamanhos dos botões estão corretos
test('button list size correctly', () => {

  const small = renderer.create(
    <ButtonList definition="are-small">
      <Button label="Small"/>
      <Button label="Small"/> 
      <Button label="Small"/>
    </ButtonList>).toJSON();
  expect(small).toMatchSnapshot();

  const medium = renderer.create(
    <ButtonList definition="are-medium">
      <Button label="Medium"/> 
      <Button label="Medium"/> 
      <Button label="Medium"/> 
    </ButtonList>).toJSON();
  expect(medium).toMatchSnapshot();

  const large = renderer.create(
    <ButtonList definition="are-large">
      <Button label="Large"/> 
      <Button label="Large"/> 
      <Button label="Large"/> 
    </ButtonList>
).toJSON();
  expect(large).toMatchSnapshot();
});

// testa se a característica fullwidth está correta
test('fullwidth  correctly', () => {

  const fullwidth = renderer.create(
    <>
    <Button  definition="is-small is-fullwidth" label="small fullwidt"/>
    <Button  definition="is-fullwidth" label="normal fullwidth"/>
    <Button  definition="is-medium is-fullwidth" label="medium fullwidth"/>
    <Button definition="is-large is-fullwidth" label="large fullwidth"/>
    </>
    ).toJSON();

  expect(fullwidth).toMatchSnapshot();
});

// testa se a característica outlined está correta
test('outlined  correctly', () => {
  const outlined = renderer.create(
    <ButtonList>
      <Button  definition="is-primary is-outlined" label="primary outlined"/>
      <Button  definition="is-link is-outlined" label=" link outlined"/>
      <Button  definition="is-info is-outlined" label="info outlined"/>
      <Button  definition="is-success is-outlined" label="success outlined"/> 
      <Button  definition="is-warning is-outlined" label="warning outlined"/>
      <Button  definition="is-danger is-outlined" label="danger outlined"/>
    </ButtonList>
    ).toJSON();

  expect(outlined).toMatchSnapshot();
});

// testa se a característica inverted está correta
test('inverted  correctly', () => {
  const inverted = renderer.create(
    <ButtonList definition="is-primary">
      <Button  definition="is-primary is-inverted" label="primary inverted"/> 
      <Button  definition="is-link is-inverted" label="link inverted"/>
      <Button  definition="is-info is-inverted" label="info inverted"/>
      <Button  definition="is-success is-inverted" label="success inverted"/> 
      <Button  definition="is-danger is-inverted" label="danger inverted"/> 
    </ButtonList>
  ).toJSON();

  expect(inverted).toMatchSnapshot();
});

// testa se a característica rounded está correta
test('rounded  correctly', () => {
  const rounded = renderer.create(
    <ButtonList>
      <Button  definition="is-primary is-rounded" label="primary rounded"/> 
      <Button  definition="is-link is-rounded" label="link rounded"/>
      <Button  definition="is-info is-rounded" label=" info rounded"/>
      <Button  definition="is-success is-rounded" label="success rounded"/>
      <Button  definition="is-warning is-rounded" label="warning rounded"/>
      <Button  definition="is-danger is-rounded" label="danger rounded"/>
    </ButtonList>
  ).toJSON();

  expect(rounded).toMatchSnapshot();
});

// testa os estados
test('state  correctly', () => {

  const hovered = renderer.create(
  <ButtonList>
    <Button  > Testando</Button>
    <Button  definition="is-primary is-hovered" label="Testando"/>
    <Button  definition="is-link is-hovered" label="Testando"/> 
    <Button  definition="is-info is-hovered" label="Testando"/> 
    <Button  definition="is-success is-hovered" label="Testando"/>
    <Button  definition="is-warning is-hovered" label="Testando"/>
    <Button  definition="is-danger is-hovered" label="Testando"/>
  </ButtonList>
    ).toJSON();
  expect(hovered).toMatchSnapshot();

  const focused = renderer.create(
  <ButtonList>
    <Button  definition="is-primary is-focused" label="primary focused"/>
    <Button  definition="is-link is-focused" label="link focused"/> 
    <Button  definition="is-info is-focused" label="info focused"/>
    <Button  definition="is-success is-focused" label="success focused"/> 
    <Button  definition="is-warning is-focused" label="warning focused"/>
    <Button  definition="is-danger is-focused" label="danger focused"/> 
  </ButtonList>).toJSON();

  expect(focused).toMatchSnapshot();

  const active = renderer.create(
    <ButtonList>
      <Button  definition="is-primary is-active" label="primary active"/> 
      <Button  definition="is-link is-active" label="link active"/>
      <Button  definition="is-info is-active" label="info active"/>
      <Button  definition="is-success is-active" label="success active"/> 
      <Button  definition="is-warning is-active" label="warning active"/> 
      <Button  definition="is-danger is-active" label="danger active"/> 
    </ButtonList>
    ).toJSON();

  expect(active).toMatchSnapshot();

});

test('loading corretly', () => {
  const loading = renderer.create(
    <ButtonList>
      <Button  definition="is-loading" label="loading"/> 
      <Button  definition="is-primary is-loading" label="loading"/>
      <Button  definition="is-link is-loading" label="loading"/>
      <Button  definition="is-info is-loading" label="loading"/>
      <Button  definition="is-success is-loading" label="loading"/>
      <Button  definition="is-warning is-loading" label="loading"/> 
      <Button  definition="is-danger is-loading" label="loading"/>
    </ButtonList>
).toJSON();

  expect(loading).toMatchSnapshot();

});

test('disabled corretly', () => {
  const disabled = renderer.create(
    <ButtonList>
      <Button  definition="is-primary" disabled={true} label="primary disabled"/> 
      <Button  definition="is-link"  disabled={true} label="link disabled"/> 
      <Button  definition="is-info" disabled={true} label="info disabled"/>
      <Button  definition="is-uccess" disabled={true} label="success disabled"/> 
      <Button  definition="is-warning" disabled={true} label="warning disabled"/>
      <Button  definition="is-danger" disabled={true} label="danger disabled"/>
    </ButtonList>
    ).toJSON();

  expect(disabled).toMatchSnapshot();

});

test('buttons with icons',()=>{
  const buttons_icons = renderer.create(
    <ButtonList>
    <Button label={<Icon icon_name="fas fa-bold"/>}/>
    
    <Button label={<Icon icon_name="fas fa-italic"/>}/>
  
    <Button label={<Icon icon_name="fas fa-underline"/>}/>
  
    <Button label={<Icon icon_name="fab fa-github">Github</Icon>}/>

    <Button definition="is-primary" label={<Icon icon_name="fab fa-twitter">Twitter</Icon>}/>
    
    <Button definition="is-success" label={<Icon icon_name="fas fa-check">Save</Icon>}/> 
  
    <Button definition="is-danger is-outlined" label={<Icon icon_name="fas fa-times" icon_right={true}>
      Delete
    </Icon>}/>

  
    <Button definition="is-small" label={<Icon icon_name="fab fa-github">
      Github
    </Icon>}/>
  
    <Button label={<Icon icon_name="fab fa-github">
      Github
    </Icon>}/>
    
  
    <Button definition="is-medium" label={<Icon icon_name="fab fa-github">
      Github
    </Icon>}/>
  
    <Button definition="is-large" label={<Icon icon_name="fab fa-github"> Github</Icon>}/>

    </ButtonList>
  
 );

 expect(buttons_icons).toMatchSnapshot();
});

