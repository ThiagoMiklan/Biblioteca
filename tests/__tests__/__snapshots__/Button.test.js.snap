// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button list size correctly 1`] = `
<div
  className="buttons are-small"
>
  <button
    className="button"
    disabled={false}
  >
    Small
  </button>
  <button
    className="button"
    disabled={false}
  >
    Small
  </button>
  <button
    className="button"
    disabled={false}
  >
    Small
  </button>
</div>
`;

exports[`button list size correctly 2`] = `
<div
  className="buttons are-medium"
>
  <button
    className="button"
    disabled={false}
  >
    Medium
  </button>
  <button
    className="button"
    disabled={false}
  >
    Medium
  </button>
  <button
    className="button"
    disabled={false}
  >
    Medium
  </button>
</div>
`;

exports[`button list size correctly 3`] = `
<div
  className="buttons are-large"
>
  <button
    className="button"
    disabled={false}
  >
    Large
  </button>
  <button
    className="button"
    disabled={false}
  >
    Large
  </button>
  <button
    className="button"
    disabled={false}
  >
    Large
  </button>
</div>
`;

exports[`buttons size correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button button is-small"
    disabled={false}
  >
    Small
  </button>
  <button
    className="button"
    disabled={false}
  >
    Normal
  </button>
  <button
    className="button button is-medium"
    disabled={false}
  >
    Medium
  </button>
  <button
    className="button button is-large"
    disabled={false}
  >
    Large
  </button>
</div>
`;

exports[`buttons with icons 1`] = `
<div
  className="buttons"
>
  <button
    className="button"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-bold"
      />
    </span>
    <span />
  </button>
  <button
    className="button"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-italic"
      />
    </span>
    <span />
  </button>
  <button
    className="button"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-underline"
      />
    </span>
    <span />
  </button>
  <button
    className="button"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fab fa-github"
      />
    </span>
    <span>
      Github
    </span>
  </button>
  <button
    className="button is-primary"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fab fa-twitter"
      />
    </span>
    <span>
      Twitter
    </span>
  </button>
  <button
    className="button is-success"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-check"
      />
    </span>
    <span>
      Save
    </span>
  </button>
  <button
    className="button is-danger is-outlined"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-times"
      />
    </span>
    <span>
      Delete
    </span>
  </button>
  <button
    className="button is-small"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fab fa-github"
      />
    </span>
    <span>
      Github
    </span>
  </button>
  <button
    className="button"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fab fa-github"
      />
    </span>
    <span>
      Github
    </span>
  </button>
  <button
    className="button is-medium"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fab fa-github"
      />
    </span>
    <span>
      Github
    </span>
  </button>
  <button
    className="button is-large"
    disabled={false}
  >
    <span
      className="icon"
    >
      <i
        className="fab fa-github"
      />
    </span>
    <span>
       Github
    </span>
  </button>
</div>
`;

exports[`color correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary"
    disabled={false}
  >
    Primary
  </button>
  <button
    className="button is-link"
    disabled={false}
  >
    Link
  </button>
  <button
    className="button is-info"
    disabled={false}
  >
    Info
  </button>
  <button
    className="button is-success"
    disabled={false}
  >
    Success
  </button>
  <button
    className="button is-warning"
    disabled={false}
  >
    Warning
  </button>
  <button
    className="button is-danger"
    disabled={false}
  >
    Danger
  </button>
</div>
`;

exports[`color secundary correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-white"
    disabled={false}
  >
    White
  </button>
  <button
    className="button is-light"
    disabled={false}
  >
    Light
  </button>
  <button
    className="button is-dark"
    disabled={false}
  >
    Dark
  </button>
  <button
    className="button is-black"
    disabled={false}
  >
    Black
  </button>
  <button
    className="button is-text"
    disabled={false}
  >
    Text
  </button>
</div>
`;

exports[`colors light correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary is-light"
    disabled={false}
  >
    primary light
  </button>
  <button
    className="button is-link is-light"
    disabled={false}
  >
    link light
  </button>
  <button
    className="button is-info is-light"
    disabled={false}
  >
    info light
  </button>
  <button
    className="button is-success is-light"
    disabled={false}
  >
    success light
  </button>
  <button
    className="button is-warning is-light"
    disabled={false}
  >
    warning light
  </button>
  <button
    className="button is-danger is-light"
    disabled={false}
  >
    danger light
  </button>
</div>
`;

exports[`disabled corretly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary"
    disabled={true}
  >
    primary disabled
  </button>
  <button
    className="button is-link"
    disabled={true}
  >
    link disabled
  </button>
  <button
    className="button is-info"
    disabled={true}
  >
    info disabled
  </button>
  <button
    className="button is-uccess"
    disabled={true}
  >
    success disabled
  </button>
  <button
    className="button is-warning"
    disabled={true}
  >
    warning disabled
  </button>
  <button
    className="button is-danger"
    disabled={true}
  >
    danger disabled
  </button>
</div>
`;

exports[`fullwidth  correctly 1`] = `
Array [
  <button
    className="button is-small is-fullwidth"
    disabled={false}
  >
    small fullwidt
  </button>,
  <button
    className="button is-fullwidth"
    disabled={false}
  >
    normal fullwidth
  </button>,
  <button
    className="button is-medium is-fullwidth"
    disabled={false}
  >
    medium fullwidth
  </button>,
  <button
    className="button is-large is-fullwidth"
    disabled={false}
  >
    large fullwidth
  </button>,
]
`;

exports[`inverted  correctly 1`] = `
<div
  className="buttons is-primary"
>
  <button
    className="button is-primary is-inverted"
    disabled={false}
  >
    primary inverted
  </button>
  <button
    className="button is-link is-inverted"
    disabled={false}
  >
    link inverted
  </button>
  <button
    className="button is-info is-inverted"
    disabled={false}
  >
    info inverted
  </button>
  <button
    className="button is-success is-inverted"
    disabled={false}
  >
    success inverted
  </button>
  <button
    className="button is-danger is-inverted"
    disabled={false}
  >
    danger inverted
  </button>
</div>
`;

exports[`loading corretly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-loading"
    disabled={false}
  >
    loading
  </button>
  <button
    className="button is-primary is-loading"
    disabled={false}
  >
    loading
  </button>
  <button
    className="button is-link is-loading"
    disabled={false}
  >
    loading
  </button>
  <button
    className="button is-info is-loading"
    disabled={false}
  >
    loading
  </button>
  <button
    className="button is-success is-loading"
    disabled={false}
  >
    loading
  </button>
  <button
    className="button is-warning is-loading"
    disabled={false}
  >
    loading
  </button>
  <button
    className="button is-danger is-loading"
    disabled={false}
  >
    loading
  </button>
</div>
`;

exports[`outlined  correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary is-outlined"
    disabled={false}
  >
    primary outlined
  </button>
  <button
    className="button is-link is-outlined"
    disabled={false}
  >
     link outlined
  </button>
  <button
    className="button is-info is-outlined"
    disabled={false}
  >
    info outlined
  </button>
  <button
    className="button is-success is-outlined"
    disabled={false}
  >
    success outlined
  </button>
  <button
    className="button is-warning is-outlined"
    disabled={false}
  >
    warning outlined
  </button>
  <button
    className="button is-danger is-outlined"
    disabled={false}
  >
    danger outlined
  </button>
</div>
`;

exports[`rounded  correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary is-rounded"
    disabled={false}
  >
    primary rounded
  </button>
  <button
    className="button is-link is-rounded"
    disabled={false}
  >
    link rounded
  </button>
  <button
    className="button is-info is-rounded"
    disabled={false}
  >
     info rounded
  </button>
  <button
    className="button is-success is-rounded"
    disabled={false}
  >
    success rounded
  </button>
  <button
    className="button is-warning is-rounded"
    disabled={false}
  >
    warning rounded
  </button>
  <button
    className="button is-danger is-rounded"
    disabled={false}
  >
    danger rounded
  </button>
</div>
`;

exports[`state  correctly 1`] = `
<div
  className="buttons"
>
  <button
    className="button"
    disabled={false}
  />
  <button
    className="button is-primary is-hovered"
    disabled={false}
  >
    Testando
  </button>
  <button
    className="button is-link is-hovered"
    disabled={false}
  >
    Testando
  </button>
  <button
    className="button is-info is-hovered"
    disabled={false}
  >
    Testando
  </button>
  <button
    className="button is-success is-hovered"
    disabled={false}
  >
    Testando
  </button>
  <button
    className="button is-warning is-hovered"
    disabled={false}
  >
    Testando
  </button>
  <button
    className="button is-danger is-hovered"
    disabled={false}
  >
    Testando
  </button>
</div>
`;

exports[`state  correctly 2`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary is-focused"
    disabled={false}
  >
    primary focused
  </button>
  <button
    className="button is-link is-focused"
    disabled={false}
  >
    link focused
  </button>
  <button
    className="button is-info is-focused"
    disabled={false}
  >
    info focused
  </button>
  <button
    className="button is-success is-focused"
    disabled={false}
  >
    success focused
  </button>
  <button
    className="button is-warning is-focused"
    disabled={false}
  >
    warning focused
  </button>
  <button
    className="button is-danger is-focused"
    disabled={false}
  >
    danger focused
  </button>
</div>
`;

exports[`state  correctly 3`] = `
<div
  className="buttons"
>
  <button
    className="button is-primary is-active"
    disabled={false}
  >
    primary active
  </button>
  <button
    className="button is-link is-active"
    disabled={false}
  >
    link active
  </button>
  <button
    className="button is-info is-active"
    disabled={false}
  >
    info active
  </button>
  <button
    className="button is-success is-active"
    disabled={false}
  >
    success active
  </button>
  <button
    className="button is-warning is-active"
    disabled={false}
  >
    warning active
  </button>
  <button
    className="button is-danger is-active"
    disabled={false}
  >
    danger active
  </button>
</div>
`;

exports[`value correctly 1`] = `
<button
  className="button"
  disabled={false}
>
  Value
</button>
`;

exports[`value correctly 2`] = `
<button
  className="button"
  disabled={false}
/>
`;
